flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPoolTest.java

```diff
@@ -661,26 +660,32 @@ public void testBlockingRequestFromMultiLocalBufferPool() throws Exception {
 			}
 			final List<MemorySegment> exclusiveSegments = globalPool.requestMemorySegments();
 			assertTrue(globalPool.isAvailable().isDone());
-			for (BufferPool localPool: localBufferPools) {
+			for (final BufferPool localPool: localBufferPools) {
 				assertTrue(localPool.isAvailable().isDone());
 			}
 
 			// blocking request buffers form local buffer pools
 			final CountDownLatch latch = new CountDownLatch(numLocalBufferPool);
 			final BlockingQueue<BufferBuilder> segmentsRequested = new ArrayBlockingQueue<>(numBuffers);
+			final AtomicReference<Throwable> cause = new AtomicReference<>();
 			for (final BufferPool localPool: localBufferPools) {
 				executorService.submit(() -> {
-					for (int num = localPoolMaxSize; num > 0; --num) {
-						ExceptionUtils.suppressExceptions(
-							() -> segmentsRequested.add(localPool.requestBufferBuilderBlocking()));
+					try {
+						for (int num = localPoolMaxSize; num > 0; --num) {
+							segmentsRequested.add(localPool.requestBufferBuilderBlocking());
+						}
+					} catch (Exception e) {
+						cause.set(e);
+					} finally {
+						latch.countDown();
 					}
-					latch.countDown();
 				});
 			}
 
 			// wait until all available buffers are requested
 			while (globalPool.getNumberOfAvailableMemorySegments() > 0) {
 				Thread.sleep(100);
+				assertNull(cause.get());
```

Comment:
We should provide some exception message here for better tracing the issue.
`assertNull(cause.get().getMessage(), cause.get())` ?

In addition this check only guarantees that there is no exceptions during non-blocking request. It might still exist the exception during blocking request process? If so we need double check it after below `latch.await();`.

Reply:
The error message will be printed out by assertNull because the ```toString``` method of the ```cause``` will be called. Besides, there is potential NPE problem if we call ```getMessage``` without checking ```cause.get() != null```.